tests/features/haart_test.py

```diff
@@ -48,7 +49,50 @@ def test_update_haart_t1(make_model, make_agent):
     # Increase cap. Agent goes on haart
     a.location.params.demographics[a.race].sex_type[a.sex_type].drug_type[
         a.drug_type
-    ].haart.prob = 2.0
+    ].haart.cap = 2.0
     a.haart.update_agent(model)
     assert a.haart.active
     assert a.haart.adherent is True
+
+    # falls off adherence
+    model.run_random = FakeRandom(1.0)
+    a.location.params.demographics[a.race].sex_type[a.sex_type].drug_type[
+        a.drug_type
+    ].haart.adherence.discontinue = 5.0
+    a.haart.update_agent(model)
+    assert a.haart.active
+    assert a.haart.adherent is False
+
+
+def test_update_haart_dx_time(make_model, make_agent):
+    model = make_model()
+    model.time = 1
+    a = make_agent(race="white")
+    model.run_random = FakeRandom(0.5)
+
+    a.hiv.active = True
+    a.hiv.dx = True
+    a.hiv.dx_time = 1  # agent dx duration is 0
+    a.haart.update_agent(model)
+    assert not a.haart.active
+
+    a.hiv.dx_time -= 10
+    a.haart.update_agent(model)
+    assert a.haart.active
+
+
+def test_update_haart_reinit(make_model, make_agent):
+    model = make_model()
+    model.time = 1
+    a = make_agent(race="white")
+
+    a.hiv.active = True
+    a.hiv.dx = True
+    a.hiv.dx_time = model.time - 1  # make duration 1, would pass if not reinit
+    a.haart.ever = True  # uses reinit prob
+    a.haart.update_agent(model)
+    assert not a.haart.active
+
+    model.run_random = FakeRandom(-1.1)
```

Comment:
why fix the random here but not before?

Reply:
So that the agent doesn't go on haart the first time but does the second. That is better achieved with changing the param tho.